### ISSUE TITLE
Dynamic Computation of Custom Metrics

### ISSUE THREAD

**Issue by percevalw** [aimhubio/aim#3295]
## 🚀 Feature

Enable the computation and display of custom metrics directly within Aim, derived from already logged data.

### Motivation

I’m finding it challenging to log many training metrics, like multiple differently weighted accuracies, directly from my PyTorch runs due to some implementation limitations. It would be great to have the option to define computed metrics from the logged data, so I can postpone some logging decisions after the run has started/ended.

### Pitch

I propose adding a feature that allows users to specify expressions for custom metrics. For example, if the logged metrics include (true positives) and support, one could define `acc = tp / support` to be calculated on the fly. This could also be useful to compute weighted average of various metrics, like F scores, or keep only some classes from a accuracy metric (e.g. acc = (tpA + tpB) / (supportA + supportB).

### Alternatives

The current workaround is to precompute these metrics in the training loop and log them directly, but this doesn't allow oversights, and re-running with new training metrics might be prohibitive it the experiment was long or expensive (my case).
